samples/socket-api/wasm-src/socket_opts.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#endif
+
+#define OPTION_ASSERT(A, B, OPTION)           \
+    if (A == B) {                             \
+        printf("%s is expected\n", OPTION);   \
+    }                                         \
+    else {                                    \
+        printf("%s is unexpected\n", OPTION); \
+        return EXIT_FAILURE;                  \
+    }
+
+int
+main(int argc, char *argv[])
+{
+    int socket_fd = 0;
+
+    printf("[Client] Create socket\n");
+    socket_fd = socket(AF_INET, SOCK_STREAM, 0);
+    if (socket_fd == -1) {
+        perror("Create socket failed");
+        return EXIT_FAILURE;
+    }
+
+// TODO: Replace this with posix sockopts when available
+#ifdef __wasi__
```

Comment:
We'd like to use the original API and translate them to Wasm. Users should not learn another set of APIs to develop Wasm application

Since timeout is an opt of a socket, `SO_SNDTIMEO`, using `setsockopt(...)` and `getsockopt(..)` is better.

Reply:
I think this is just an intermediate step and it will be replaced with `getsockopt()` and `secksockopt()` once we have it implemented in `wasi_socket_ext.c`, is that right, @cimacmillan ?

Reply:
That was my original plan, but I'll raise a revision for it as part of this PR if that's preferred